packages/notebook-extension/src/index.ts

```diff
@@ -1957,7 +1962,9 @@ function activateNotebookHandler(
     caption: trans.__('Create a new notebook'),
     icon: args => (args['isPalette'] ? undefined : notebookIcon),
     execute: args => {
-      const cwd = (args['cwd'] as string) || (defaultBrowser?.model.path ?? '');
+      const currentBrowser =
+        filebrowserFactory?.tracker.currentWidget ?? defaultBrowser;
```

Comment:
If this pattern is needed across the board to support custom drives, we should probably document it for extension authors. I would suggest either pushing a small docs update or opening an issue to track doing it later.

Reply:
Yes, just opened https://github.com/jupyterlab/jupyterlab/issues/15939.

Also there are still a few fixes needed for drives in general, such as implementing support for checkpoints in the `ContentsManagerMock`, which could be done as part of https://github.com/jupyterlab/jupyterlab/issues/15930 (and other linked issues).